.github/workflows/continuous-integration-workflow.yml

```
@@ -19,3 +19,48 @@ jobs:
 
     - name: Run Docker
       run: docker run -e FLASK_APP=${{ secrets.FLASK_APP_X}} -e FLASK_ENV=${{ secrets.FLASK_ENV_X}} -e SECRET_KEY=${{ secrets.SECRET_KEY_X }} todo-app:devtest
+
+  job-two:
+    name: image build & push 
+    runs-on: ubuntu-latest
+    needs: job-one
+    steps:
```

Although this needs `job-one`, it will run on every push to any branch which we don't want. We would benefit from adding an `if` gate here (and on the deploy-to-heroku step too):
```suggestion
    needs: job-one
    if: github.ref == 'refs/heads/master'
    steps:
```